### PR TITLE
Enhance match struct 1

### PR DESCRIPTION
**Pull request by ketkarameya**
This is the same PR as https://github.com/uber/piranha/pull/321/files 
Due to some git issue (I accidently merged it into a branch, not into master), had to remake this PR. 

image

### CODE REVIEW

File: src/models/source_code_unit.rs [hunk 1]
```diff
@@ -468,10 +468,15 @@ impl SourceCodeUnit {
...
-      Match::new(new_deleted_range, edit.p_match().matches().clone()),
+      Match::new(
+        self.code()[new_deleted_range.start_byte..new_deleted_range.end_byte].to_string(),
+        new_deleted_range,
+        edit.p_match().matches().clone(),
+      ),
       edit.replacement_string().to_string(),
       edit.matched_rule().to_string(),
     );
+    // new_deleted_range
```
Comment (lazaroclapp) [inline src/models/source_code_unit.rs:479]: Um? What's this comment? Wasn't in the previous PR. Typo?

**Review by lazaroclapp — COMMENTED** [review]
Still LGTM, but remove the extra comment?

File: src/models/source_code_unit.rs (hunk 1 above)
Comment (ketkarameya) [reply src/models/source_code_unit.rs]: addressed.

**Review by lazaroclapp — APPROVED** [review]
LGTM!

**Pull request merged**